corehq/apps/locations/permissions.py

```diff
@@ -174,19 +176,18 @@ def user_can_edit_location(user, sql_location, project):
 
 
 def user_can_view_location(user, sql_location, project):
-    if (user.is_domain_admin(project.name) or
-            not project.location_restriction_for_users):
+    if user_can_edit_any_location(user, project):
         return True
 
-    user_loc = get_user_location(user, sql_location.domain)
+    user_loc = user.get_location_ids(sql_location.domain)
 
     if not user_loc:
         return True
 
     if user_can_edit_location(user, sql_location, project):
         return True
 
-    return sql_location.location_id in user_loc.lineage
+    return sql_location.location_id in user_loc
```

Comment:
This changes the meaning of the function - with the old behavior, users could see view location above theirs.  This change makes it so they can only see locations they're assigned to.